controller/api/views.py

```diff
@@ -277,6 +282,17 @@ class ConfigViewSet(ReleasableViewSet):
     model = models.Config
     serializer_class = serializers.ConfigSerializer
 
+    def create(self, request, **kwargs):
+        # Guard against overlapping config changes, using a filesystem lock so that
+        # multiple controller processes can be coordinated.
+        lockfile = os.path.join(tempfile.gettempdir(), kwargs['id'])  # e.g., /tmp/violet-valkyrie
```

Comment:
given it is config related, or more precisely build / deploy then shouldn't we append that so we can also have other locks for diff operations under the same namespace?